scipy/stats/_stats_mstats_common.py

```diff
@@ -22,6 +22,8 @@ def linregress(x, y=None):
         If only x is given (and y=None), then it must be a two-dimensional
         array where one dimension has length 2.  The two sets of measurements
         are then found by splitting the array along the length-2 dimension.
+        In the case where y=None and x is a 2x2 array, the splitting will be
+        done along the rows, resulting in the sets x[0] and x[1].
```

Comment:
Nit: Maybe rephrase: ... along the rows, i.e., `linregress(x)` is equivalent to `linregress(x[0], x[1])`.

Also, I'd suggest an immediately obvious example, e.g.
```
>>> x = np.array([[0, 1], [0, 2]])
>>> r = linregress(x)
>>> r.slope, r.intercept
(2, 0)
```

Reply:
Thanks @ev-br. Do prefer that line in the parameters section or in the notes?
Also, would that example be the last one in the Examples section?

Reply:
`Notes` are too deep down, I'd put the phrase right there when tge parameter is first introduced.